whistle/lib/HomeScreen.dart

```diff
@@ -0,0 +1,369 @@
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:whistle/models/constants.dart';
+import 'models/playlist.dart';
+import 'models/song.dart';
+
+class HomeScreen extends StatefulWidget {
+  const HomeScreen({Key? key}) : super(key: key);
+
+  @override
+  _HomeScreenState createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  String buttonName = 'Click';
+  bool isClicked = false;
+  int _selectedIndex = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    Size size = MediaQuery.of(context).size;
+
+    return Scaffold(
+      backgroundColor: Colors.white,
+      appBar: AppBar(
+        title: Text(
+          'Whistle',
+          style: Theme.of(context)
+              .textTheme
+              .headline4
+              ?.copyWith(fontWeight: FontWeight.bold, color: kPrimaryColor),
+        ),
+        elevation: 0,
+        backgroundColor: Colors.transparent,
+        actions: [
+          Padding(
+            padding: const EdgeInsets.all(10.0),
+            child: Icon(
+              Icons.account_circle,
+              color: kPrimaryColor,
+              size: 30,
+            ),
+          )
+        ],
+      ),
+      body: Column(
+        children: [
+          Container(
+            height: size.height * 0.70,
+            child: Row(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [_buildNavigationRail(), _buildPlayListAndSongs(size)],
+            ),
+          ),
+          _buildCurrentPlayingSong(size),
+          _buildBottomBar(size),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildBottomBar(Size size) {
+    return Container(
+      height: size.height * 0.065,
```

Comment:
Same here, hard coded double may leave an issue on smaller/bigger screen, need to test for different phones and screens i guess